tabs/tab_survival.py

```
@@ -268,244 +241,484 @@ def surv_ui(namespace: str) -> ui.TagChild:
+            for kw in tx_keywords:
+                matched = [c for c in cols if kw in c.lower()]
+                if matched:
+                    default_tx = matched[0]
+                    break
+            
+            # Fallback: ถ้าไม่เจอ keyword เลย ให้เลือกคอลัมน์แรกสุด
+            if default_tx == "Select..." and cols:
+                default_tx = cols[0]
+
+            # 5. Detect subgroup Variables (ตามลำดับความสำคัญใน event_keywords)
+            subgr_keywords = ['comorbid', 'comorb', 'group', 'control', 'contr', 'ctr' 'dz', 'lab', 'diag', 'sex', 'age']
```

Ruff (0.14.10)

Implicitly concatenated string literals on one line

Combine string literals

(ISC001)